deltachat-jsonrpc/src/api/mod.rs

```diff
@@ -1336,6 +1337,40 @@ impl CommandApi {
         .await
     }
 
+    /// Starts to provide the backup for remote devices to retrieve.
+    ///
+    /// Can be cancelled by stopping the ongoing process.  Success or failure can be tracked
+    /// via the `ImexProgress` event which should either reach `1000` for success or `0` for
+    /// failure.
+    ///
+    /// This **stops IO**.  After completion `start_io` must be called to restart IO.
+    ///
+    /// Returns the QR code as a rendered SVG image.
+    async fn provide_backup(&self, account_id: u32, path: String) -> Result<String> {
```

Comment:
@Simon-Laux @link2xt Question about how a JSON-RPC API is allowed to behave.  This starts something in the background.  However knowing it has finished is via watching for events.  That's pretty disconnected, is that fine for JSON-RPC?

The Rust and C APIs give you an object that you can then use, but here I don't think that's a pattern?

Reply:
You could always return a serialisable handle, like `{context_id: u32, provider_id: u32}` and look it back up.  This is not really different from your suggestion anyway, because in both cases you need to store the handle in the context so you can find it again.

I guess the benefit of introducing a `provider_id`-like thing for the handle is that you can't accidentally get the QR code for a previous provider.

So ultimately the question is whether we change this so we store the `BackupProvider` in the `Context` or not.  I gather from this that you'd prefer doing that as it allows a blocking `provide_backup` call without having to look at events to figure out when it finishes?

Also, just for completeness of my understanding: this means json-rpc is happy to handle multiple concurrent calls even when some of these are taking a long time and are blocking?